src/systemctl/sysv-compat.c

```diff
@@ -65,38 +65,65 @@ int parse_shutdown_time_spec(const char *t, usec_t *ret) {
 
                 *ret = now(CLOCK_REALTIME) + USEC_PER_MINUTE * u;
         } else {
-                char *e = NULL;
-                long hour, minute;
+                char *hour_s;
+                char *min_s;
+                char *colon_position;
+                unsigned long hour, minute;
                 struct tm tm = {};
                 time_t s;
                 usec_t n;
+                bool relative = false;
+                int r;
 
-                errno = 0;
-                hour = strtol(t, &e, 10);
-                if (errno > 0 || *e != ':' || hour < 0 || hour > 23)
+                if (t[0] == '+') {
+                        relative = true;
+                        t++;
+                }
+
+                colon_position = strchr(t,':');
+                if (colon_position == NULL)
+                        return -EINVAL;
+
+                hour_s = strndup(t, colon_position - t);
+                min_s = strndup(colon_position + 1, 2);
```

Comment:
ditto

Reply:
that said, there's no point in specifiying 2 as length here. just use the remaining string as is. otherwise you'll parse 00:00abcde as valid time, but it shouldn't be

Reply:
and there's no reason to copy the remainder, you can just use the suffix ptr as is